### PR TITLE
Ensure ButtonGroup children have appropriate margins

### LINKED ISSUE #528: Button group + Safari = unexpected left/right margins
Came here after a talk at React Rally :) I searched the issue queue and I wasn't able to find anything right off hand, but if this has existed before then my apologies.

In Safari, "button group" buttons have what appear to be an unexpected `2px` margin left /right. See attached screenshots. 

Screen Shot 2019-08-23 at 2 17 17 PM
Screen Shot 2019-08-23 at 2 17 27 PM

### PR DESCRIPTION
**Pull request by BinaryMuse**
Safari has a 2px margin by default on `button`s, so this is necessary to get `ButtonGroup` to render correctly on Safari.

**Before**

image

**After**

image

Fixes #528

#### Merge checklist
- [x] Changed ~base branch to release branch~ version number
- [x] Add or update TypeScript definitions (`index.d.ts`) if necessary
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@primer/components",
-  "version": "14.3.1",
+  "version": "14.3.2",
```
Comment (emplums) [inline package.json]: This is new as of today, but since there's been so much activity in this repo lately (yay! 🎉) I think we should start doing release tracking PRs again to avoid versioning errors with so many things being shipped! If you want to just revert this change and then change the base branch to `release-14.4.0` we should be good to go!

Comment (BinaryMuse) [reply package.json]: Got it!
